File: 프로그래머스/lv2/84512. 모음 사전/모음 사전.py
result = 0
cnt = 0
def solution(word):
    list = []
    def dfs():
        global cnt, result
        if "".join(list) == word:
            result = cnt
            return

        if len(list) == 5:
            return

        alpha = ['A', 'E', 'I', 'O', 'U']

        for i in range(5):
            cnt += 1
            list.append(alpha[i])
            dfs()
            list.pop()

    dfs()
    return result

File: 프로그래머스/lv2/84512. 모음 사전/모음 사전.java
class Solution {
    int answer, cnt;
    public void findWordIndex(StringBuilder sb, String word){
        if(sb.toString().equals(word)){
            answer = cnt;
            return;
        }
        if(sb.length() == 5) return;

            
        char[] words = {'A', 'E', 'I', 'O', 'U'};
      
        for(int i=0; i<5; i++){ // words idx
            sb.append(words[i]);
            cnt++;
            findWordIndex(sb, word);
            sb.deleteCharAt(sb.length() - 1);
        }
        
    }
    public int solution(String word) {
        findWordIndex(new StringBuilder(), word);
        return answer;
    }
}